memory/concurrent_arena.h

```diff
@@ -49,7 +49,9 @@ class ConcurrentArena : public Allocator {
 
   char* Allocate(size_t bytes) override {
     return AllocateImpl(bytes, false /*force_arena*/,
-                        [=]() { return arena_.Allocate(bytes); });
+      [ROCKSDB_THIS_LAMBDA_CAPTURE]() {
```

Comment:
Unfortunate, but I guess we need it to avoid newer C++ and older C++ disagreeing on which is worth a warning

Reply:
@pdillinger Honestly, it we can discusse about it. I have another way to fix it :
- Use the [=, this] everywhere, which is the new way to handle such pattern
- Ignore warning on pre std 20 compilers

Reply:
At this time, we write C++11 with warnings as errors. We aren't going to use a patchwork of trying to ignore this specific warning for various compilers in our default configuration. (I haven't even found a way to turn it off in gcc while keeping -Werror.)

Exactly which C++20 compiler are you getting an error with? I tried latest gcc, clang and msvc on godbolt.org and was not able to get even a warning. Can't you suppress in the C++20 case? (deprecated != removed)

https://godbolt.org/z/8NCWeL
https://godbolt.org/z/tLtxc_
https://godbolt.org/z/gPKt3s (see also https://developercommunity.visualstudio.com/content/problem/749985/msvc-1922-emitts-c4626-warning-on-simple-lambdas-w.html)

Reply:
@pdillinger 
With gcc 9 -std=c++11: https://godbolt.org/z/Z2DrYm
With gcc 9 -std=c++2a: https://godbolt.org/z/knfpAh

With "-Wno-error=deprecated" it may fix the error and leave the warning, but I don't like it...
 
On the other hand, I think there is a better (and easier) way : just dop the default capture and capture explicitly. It is cleaner and should work everywhere.